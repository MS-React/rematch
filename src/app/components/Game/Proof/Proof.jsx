import React from 'react';
import PropTypes from 'prop-types';

const Proof = ({ proof }) => (
  <div key="game-proof" className="game-proof">
    {`${proof.equation} = ?`}
  </div>
);

Proof.propTypes = {
  proof: PropTypes.shape({
    equation: PropTypes.string,
    result: PropTypes.number,
  }).isRequired,
};

export default Proof;
