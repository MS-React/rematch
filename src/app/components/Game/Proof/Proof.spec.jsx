import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import Proof from './Proof';

describe('Proof', () => {
  let wrapper;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      proof: {
        equation: '1 + 1',
      },
    };
    wrapper = shallow(<Proof {...defaultProps} />);
  });

  it('should show user the summary reportto the user', () => {
    expect(renderer.create(wrapper).toJSON())
      .toMatchSnapshot();
  });
});
