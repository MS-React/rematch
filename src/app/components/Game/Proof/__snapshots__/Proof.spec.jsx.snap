// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Proof should show user the summary reportto the user 1`] = `
<div
  className="game-proof"
>
  1 + 1
   
  = ?
</div>
`;
